app.js

```diff
@@ -11,3 +11,82 @@ document.querySelectorAll('.nav-link').forEach((n) => n.addEventListener('click'
   navMenu.classList.remove('active');
   document.querySelector('body').classList.toggle('active');
 }));
+// Get Data
+const form = document.querySelector('form');
+const nameInput = document.querySelector('.name');
+const email = document.querySelector('.email');
+const message = document.querySelector('#message');
+const success = document.querySelector('#success');
+const errorNodes = document.querySelectorAll('.error');
+form.addEventListener('submit', (ev) => {
+  function validateForm() {
+    // clear error / success messages
+    function clearMessage() {
+      for (let i = 0; i < errorNodes.length; i += 1) {
+        errorNodes[i].innerText = '';
+      }
+      success.innerText = '';
+    }
+    clearMessage();
+    let errorFlag = false;
+    if (nameInput.value.length < 1) {
+      ev.preventDefault();
+      errorNodes[0].innerText = 'Name can not be blank';
+      errorFlag = true;
+    }
+    // check if email is valid
+    function emailIsValid(email) {
+      const pattern = /\S+@\S+\.\S+/;
+      return pattern.test(email);
+    }
+    if (!emailIsValid(email.value)) {
+      ev.preventDefault();
+      errorNodes[1].innerText = 'Invalid Email Address';
+      errorFlag = true;
+    }
+    if (message.value.length < 5) {
+      ev.preventDefault();
+      errorNodes[2].innerText = 'Please Enter Message';
+      errorFlag = true;
+    }
+    if (!errorFlag) {
+      success.innerText = 'Success!';
+    }
+  }
+  validateForm();
+});
+
+const usrD = {};
+function saveD(usrD) {
+  const fName = document.querySelector('.name').value;
+  const fEmail = document.querySelector('.email').value;
+  const fMessage = document.querySelector('#message').value;
+  usrD = {
+    name: fName,
+    email: fEmail,
+    message: fMessage,
+  };
+  localStorage.setItem('user', JSON.stringify(usrD));
+  let dataRecord = [];
+  dataRecord = JSON.parse(localStorage.getItem('dUser'));
+  dataRecord.push({
+    name: fName,
+    email: fEmail,
+    message: fMessage,
+  });
+  localStorage.setItem('dUser', JSON.stringify(dataRecord));
+}
+
+form.addEventListener('change', () => {
+  saveD(usrD);
+});
+
+window.addEventListener('load', () => {
```

Comment:
- [ ] Kindly make the contact form information persistent to page reloads, currently, if users don't submit after inputting the data goes away 
- [ ] Please ensure that the data is saved on the browser when a previous user visits and it is populated with previous inputs, if no inputs were found it should remain empty 👍🏿

Reply:
> * [ ]  Kindly make the contact form information persistent to page reloads, currently, if users don't submit after inputting the data goes away
> * [ ]  Please ensure that the data is saved on the browser when a previous user visits and it is populated with previous inputs, if no inputs were found it should remain empty 👍🏿

- [ ] As mentioned by the previous reviewer, please make sure that the data is saved, when the user has not submitted the form yet, something like, when they enter the data, then reload the page, they should see entered data persisted, saved in the field, then they can continue filling from were they left... but what your site does is, it saves the data only after the user has submitted the form. please arrange this and you will be good to go.